src/components/query-form/query-form.js

```diff
@@ -111,6 +114,43 @@ const QueryForm = props => {
     }
   }
 
+  let dataObj, data, filename, tempLink
+  const queryDataDownloadLink = (fileType) => {
+    dataObj = downloadLinkData(queryData, fileType)
+    data = new Blob([dataObj], {type: fileType})
+    filename = fileType == "text/csv" ? "query_results.csv" : "query_results.json"
+
+    tempLink = document.createElement('a')
+    tempLink.href = window.URL.createObjectURL(data)
+    tempLink.setAttribute('download', filename)
+    tempLink.click()
+  }
+
+  const downloadLinkData = (queryData, fileType) => {
+    if(fileType == "text/csv") return queryDataToCSV(queryData)
```

Comment:
How does this handle hierarchical data? (i.e. truly hierarchical, not embedded json)

Reply:
that is a good question